src/StorageFileHelper.php

```diff
@@ -47,11 +53,26 @@ public static function listFiles(ClientWrapper $clientWrapper, ListFilesOptions
                     return the same client in this case (practically one is Client and one is BranchAwareClient class,
                     but they work the same in this case))
         */
+        $listFilesOptions = (new ListFilesOptions())
+            ->setQuery($tagsToQueryProcessor->toQuery($tags))
+            ->setLimit($limit);
+
         $sourceBranchId = $clientWrapper->getBranchId();
         $files = $clientWrapper->getTableAndFileStorageClient()->listFiles($listFilesOptions);
         if (!$files && $clientWrapper->isDevelopmentBranch() &&
             $clientWrapper->getClientOptionsReadOnly()->useBranchStorage()
         ) {
+            // change tags to use the default branch
+            $tags = new Tags(
+                $clientWrapper->getDefaultBranch()->id,
```

Comment:
Toto je ta důležitá změna, kvůli který celý ten kolotoč

Reply:
Jop